src/notation/view/partlistmodel.cpp

```diff
@@ -211,70 +163,64 @@ void PartListModel::removePart(int partIndex)
 
 void PartListModel::doRemovePart(int partIndex)
 {
-    if (!isNotationIndexValid(partIndex)) {
+    if (!isExcerptIndexValid(partIndex)) {
         return;
     }
 
+    bool isCurrentContation = context()->currentNotation() == m_excerpts[partIndex]->notation();
+
     beginRemoveRows(QModelIndex(), partIndex, partIndex);
-    m_notations[partIndex]->setOpened(false);
-    m_notations.removeAt(partIndex);
+
+    masterNotation()->notation()->undoStack()->prepareChanges();
+    masterNotation()->removeExcerpts({ m_excerpts[partIndex] });
```

Comment:
Better to move prepareChanges() and commitChanges() to the removeExcerpts implementation